linode/manager
# refactor: [M3-8981] - User Preferences optimization with selector pattern (Part 1)

## Description 📝

- Implements the "selector pattern" for our `usePreferences` hook
  - This allows us to "subscribe" to certain parts of the preferences object, which help us prevent unnecessary re-renders when an unrelated preference is updated 🏎️ 
  - See [this post](https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option) for a description of this React Query pattern

> [!note]
> This is just part 1, which introduces the pattern and uses it in a small number of places. I will follow up this PR with a part  2 (and maybe 3) to implement the pattern throughout the app. 

## Preview 📷

> [!important]
> No UI changes expected

## How to test 🧪


- Checkout this PR (or use a preview link)
- Check the code
  - Verify the new pattern is intuitive
  - Check for code quality
- To confirm this does not break anything
  - Verify the primary navigation collapsed state is correctly saved in user preferences
  - Verify the primary navigation collapsed state persists on related


<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
